js/InputReader.js

```diff
@@ -0,0 +1,64 @@
+export class InputReader {
+    #input
+
+    constructor(input) {
+        if (typeof(input) !== "string") {
+            throw new TypeError("Expected input to be a string.")
+        }
+
+        this.#input = input
+    }
+
+    /**
+     * Iterates through the input string and yields all contained tokens.
+     *
+     * A token is either a Number representing an operand,
+     * or a string representing an operator or a bracket.
+     * E.g. the input "(3+45)" yields: "(", 3, "+", 45, ")"
+     */
+    *iterateTokens() {
+        for (let i = 0; i < this.#input.length; i++) {
+            if (InputReader.#isNumber(this.#input[i])) {
+                let number = InputReader.#completeNumberParser(this.#input, i)
+                i += InputReader.#lengthOfNumber(number) - 1
+                yield parseInt(number)
+            } else if (this.#input[i] === " ") {
+                continue
+            } else {
+                yield this.#input[i]
+            }
+        }
+    }
+
+    static #isNumber(char) {
+        const numbers = [ "0", "1", "2", "3", "4", "5", "6", "7", "8", "9" ]
+        for (const i of numbers) {
+            if (char === i) {
+                return true
+            }
+        }
+
+        return false
+    }
+
+    static #completeNumberParser(input, i) {
```

Comment:
Is this function completing the NumberParser? ))

Also, you don't have to take `input` as an parameter, as it is a class variable and can be accessed inside all class methods without explicitly being passed down. In order for that to work you need to make this method non-static and change the caller from `InputReader.#completeNumberParser(this.#input, i)` to `this.#completeNumberParser(i)`.